posthog/hogql/compiler/bytecode.py

```diff
@@ -187,15 +192,21 @@ def visit_not(self, node: ast.Not):
     def visit_compare_operation(self, node: ast.CompareOperation):
         operation = COMPARE_OPERATIONS[node.op]
         if operation in [Operation.IN_COHORT, Operation.NOT_IN_COHORT]:
-            cohort_name = ""
-            if isinstance(node.right, ast.Constant):
-                if isinstance(node.right.value, int):
-                    cohort_name = f" (cohort id={node.right.value})"
+            if self.cohort_membership_supported:
+                if operation == Operation.IN_COHORT:
+                    return self.visit(ast.Call(name="inCohort", args=[node.right]))
                 else:
-                    cohort_name = f" (cohort: {str(node.right.value)})"
-            raise QueryError(
-                f"Can't use cohorts in real-time filters. Please inline the relevant expressions{cohort_name}."
-            )
+                    return self.visit(ast.Call(name="notInCohort", args=[node.right]))
```

Comment:
@mariusandra I think you'll need to check this out - im less familiar with the bytecode stuff. But I would have thought we couldn't do this without an `inCohort()` and `notInCohort()` implementation?